OMCompiler/Compiler/BackEnd/CommonSubExpression.mo

```diff
@@ -885,11 +884,22 @@ algorithm
             exarray := ExpandableArray.update(id, cseEquation, exarray);
           end if;
         else
-          print("This should never appear\n");
+          Error.addMessage(Error.GENERIC_ELAB_EXPRESSION, {ExpressionDump.dumpExpStr(inExp, 0) + " This should never happen, Error in wrapFunctionCalls_analysis3. Trying to recover."});
         end if;
       end if;
     then true;
 
+    /*
+      check arguments of noEvent for functions but ignore noEvent itself.
+      ticket #5771
+    */
+    case DAE.CALL(Absyn.IDENT("noEvent"),{DAE.RELATION(e,_,e2,_,_)}) algorithm
+      (_, outTuple) := Expression.traverseExpTopDown(e, wrapFunctionCalls_analysis3, inTuple);
+      (_, outTuple) := Expression.traverseExpTopDown(e2, wrapFunctionCalls_analysis3, outTuple);
+      cont := false;
+      return;
+    then true;
```

Comment:
@kabdelhak What does the return statement in this case?

Reply:
Hmmm nothing i guess. I just copied the behaviour from the DAE.IFEXP() case. I could just return false.

Reply:
Ah i see, i tried just returning false and it failed. The return statement prevents the last line in this function from being executed. ` outTuple := (HT, exarray, cseIndex, index, functionTree);`

Reply:
How would it be different to `fail()`?

Reply:
Ahh, it is not failing but returning with an undefined value for `outTuple`, right?

Reply:
Never mind - now I got it!

Reply:
Ok perfect. :D